packages/snaps-controllers/src/snaps/endowments/signature-insight.ts

```diff
@@ -0,0 +1,150 @@
+import type {
+  PermissionSpecificationBuilder,
+  EndowmentGetterParams,
+  ValidPermissionSpecification,
+  PermissionValidatorConstraint,
+  PermissionConstraint,
+  CaveatSpecificationConstraint,
+  Caveat,
+} from '@metamask/permission-controller';
+import { PermissionType, SubjectType } from '@metamask/permission-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import { SnapCaveatType } from '@metamask/snaps-utils';
+import type { Json, NonEmptyArray } from '@metamask/utils';
+import { assert, hasProperty, isObject, isPlainObject } from '@metamask/utils';
+
+import { SnapEndowments } from './enum';
+
+const permissionName = SnapEndowments.SignatureInsight;
+
+type SignatureInsightEndowmentSpecification = ValidPermissionSpecification<{
+  permissionType: PermissionType.Endowment;
+  targetName: typeof permissionName;
+  endowmentGetter: (_options?: EndowmentGetterParams) => undefined;
+  allowedCaveats: Readonly<NonEmptyArray<string>> | null;
+  validator: PermissionValidatorConstraint;
+}>;
+
+/**
+ * `endowment:signature-insight` returns nothing; it is intended to be used as a flag
+ * by the extension to detect whether the snap has the capability to show information on the signature confirmation screen.
+ *
+ * @param _builderOptions - Optional specification builder options.
+ * @returns The specification for the signature-insight endowment.
+ */
+const specificationBuilder: PermissionSpecificationBuilder<
+  PermissionType.Endowment,
+  any,
+  SignatureInsightEndowmentSpecification
+> = (_builderOptions?: unknown) => {
+  return {
+    permissionType: PermissionType.Endowment,
+    targetName: permissionName,
+    allowedCaveats: [SnapCaveatType.SignatureOrigin],
+    endowmentGetter: (_getterOptions?: EndowmentGetterParams) => undefined,
+    validator: ({ caveats }) => {
+      if (
+        (caveats !== null && caveats?.length > 1) ||
+        (caveats?.length === 1 &&
+          caveats[0].type !== SnapCaveatType.SignatureOrigin)
+      ) {
+        throw rpcErrors.invalidParams({
+          message: `Expected a single "${SnapCaveatType.SignatureOrigin}" caveat.`,
+        });
+      }
+    },
+    subjectTypes: [SubjectType.Snap],
+  };
+};
+
+export const signatureInsightEndowmentBuilder = Object.freeze({
+  targetName: permissionName,
+  specificationBuilder,
+} as const);
+
+/**
+ * Validates the type of the caveat value.
```

Comment:
```suggestion
 * Validate the type of the caveat value.
```

Reply:
We use "Validates" everywhere else, "Validate the type of the caveat value" sounds more like a directive than a description of the function.